back/lib/tasks/single_use/20250421_update_published_participants_data.rake

```diff
@@ -0,0 +1,117 @@
+namespace :single_use do
```

Comment:
This looks like a pretty major data migration.

I would think about making it possible to run without saving any changes (a dry run), to check first if it runs well without erroring half way though on production.

I usually do this by requiring a specific argument value to execute (i.e. actually make changes), and otherwise default to dry run.

For an example, (in Project Library) see: https://github.com/CitizenLabDotCo/cl2-library/blob/master/lib/tasks/data_manipulation/correct_inconsistent_marked_deleted_at.rake

Reply:
That's pretty clever. I will add a dry run that instead of making changes validates the published data structure, and log something if there is something unexpected before running the version that does change stuff 👍